helper/schema/grpc_provider.go

```diff
@@ -607,12 +607,37 @@ func (s *GRPCProviderServer) ConfigureProvider(ctx context.Context, req *tfproto
 	// request scoped contexts, however this is a large undertaking for very large providers.
 	ctxHack := context.WithValue(ctx, StopContextKey, s.StopContext(context.Background()))
 
+	// NOTE: This is a hack to pass the deferral_allowed field from the Terraform client to the
+	// underlying (provider).Configure function, which cannot be changed because the function
+	// signature is public. (╯°□°)╯︵ ┻━┻
+	s.provider.deferralAllowed = configureDeferralAllowed(req.ClientCapabilities)
+
 	logging.HelperSchemaTrace(ctx, "Calling downstream")
 	diags := s.provider.Configure(ctxHack, config)
 	logging.HelperSchemaTrace(ctx, "Called downstream")
 
 	resp.Diagnostics = convert.AppendProtoDiag(ctx, resp.Diagnostics, diags)
 
+	if s.provider.providerDeferred != nil {
+		// Check if a deferred response was incorrectly set on the provider. This would cause an error during later RPCs.
+		if !configureDeferralAllowed(req.ClientCapabilities) {
```

Comment:
Nit: Can this just check `!s.provider.deferralAllowed` here?

Reply:
I'm assuming you're just referring to the 2nd `if`, which can definitely be replaced by that -> 05985636e2f256b17c2f7149a624c4e084ea57b8 ⚡ 

The initial `s.provider.providerDeferred != nil` is still needed because we only want to raise a diagnostic if a deferral is returned by the provider incorrectly (since it won't be returned to `terraform-plugin-go`, this is really just preventing an error in later RPCs)